src/feature/dirauth/dirvote.c

```diff
@@ -3200,6 +3201,25 @@ dirvote_add_vote(const char *vote_body, const char **msg_out, int *status_out)
     goto err;
   }
 
+  if (!time_posted) { /* I imported this one myself */
+    log_notice(LD_DIR, "Retrieved vote from %s.", vi->address);
+  }
+
+  /* Check if we received it, as a post, after the cutoff when we
+   * start asking other dir auths for it. If we do, the best plan
+   * is to discard it, because using it greatly increases the chances
+   * of a split vote for this round (some dir auths got it in time,
+   * some didn't). */
+  if (time_posted && time_posted > voting_schedule.fetch_missing_votes) {
+    char tbuf1[ISO_TIME_LEN+1], tbuf2[ISO_TIME_LEN+1];
+    format_iso_time(tbuf1, time_posted);
+    format_iso_time(tbuf2, voting_schedule.fetch_missing_votes);
+    log_warn(LD_DIR, "Rejecting vote from %s received at %s; "
+             "our cutoff for received votes is %s", vi->address, tbuf1, tbuf2);
+    *msg_out = "Vote received too late, would be dangerous to count it";
```

Comment:
Should this be a more actionable warning? It could suggest checking your clock is skewed, and checking whether your CPU is hosed. If neither of those is true, then we have a bug somewhere.

Reply:
Here are the different possible causes:
* skewed clock
* CPU load
* network load
Late votes can be caused by these issues on the local or remote authority.

I edited the log message in d8c719efd96bf1cbc584fe66c2968b9e18fefe96.